source/GameData.cpp

```diff
@@ -1068,8 +1080,30 @@ void GameData::LoadFile(const string &path, bool debugMode)
 		}
 		else if(key == "shipyard" && node.Size() >= 2)
 			shipSales.Get(node.Token(1))->Load(node, ships);
-		else if(key == "start")
-			startConditions.Load(node);
+		else if(key == "start" && node.Size() >= 2)
+		{
+			// This copies the StartConditions. We're avoiding using the pointer
+			// returned by Get in case the newly updated start conditions turn out to be invalid
+			StartConditions thisStart = (*namedStartConditions.Get(node.Token(1)));
+			thisStart.Load(node);
+			
+			if (!thisStart.Valid())
```

Comment:
@tehhowch How should I do `node.PrintTrace` (in case the start conditions are invalid) after the node and its parents go out of scope?

Reply:
> How should I do `node.PrintTrace` (in case the start conditions are invalid) after the node and its parents go out of scope?

You don't. As mentioned, we can't actually check if something is valid until everything is loaded. Otherwise you might have a scenario where a start references a system or planet that hasn't been defined yet, erroneously marking the start as invalid.

Looking to #3462, not everything that is invalid necessarily gets printed to the errors file. In our case here, we only want to use the IsValid function to exclude showing those starts which aren't valid, not print out a list of all invalid starts.

Reply:
@Amazinite I've decided to log the error using Files::LogError (which doesn't need us to store the DataNode) instead not printing anything, because I think it would be useful for a plugin developer in case they don't realize what's wrong with the scenario they made. Files::LogError is non-critical, so logging it won't do any harm.